relays/bin-substrate/src/messages_source.rs

```diff
@@ -245,10 +252,40 @@ where
 	}
 
 	async fn estimate_confirmation_transaction(&self) -> P::SourceChainBalance {
-		num_traits::Zero::zero() // TODO: https://github.com/paritytech/parity-bridges-common/issues/997
+		self.client
```

Comment:
Actually, this may be called from `fn new()` and then just returned here, but it'll add additional assumptions - like that call weights may not be updated while relayer is running.